db/schema.go

```diff
@@ -128,31 +127,37 @@ func (db *db) patchSchema(ctx context.Context, txn datastore.Txn, patchString st
 		return err
 	}
 
-	var newDescriptionsByName map[string]client.CollectionDescription
+	var newSchemaByName map[string]client.SchemaDescription
 	decoder := json.NewDecoder(strings.NewReader(string(newDescriptionJson)))
 	decoder.DisallowUnknownFields()
-	err = decoder.Decode(&newDescriptionsByName)
+	err = decoder.Decode(&newSchemaByName)
 	if err != nil {
 		return err
 	}
 
 	newCollections := []client.CollectionDefinition{}
-	newSchemaByName := map[string]client.SchemaDescription{}
-	for _, desc := range newDescriptionsByName {
-		col, err := db.newCollection(desc, desc.Schema)
+	for _, schema := range newSchemaByName {
+		if schema.Name == "" {
+			return ErrSchemaNameEmpty
+		}
+
+		collectionDescription, ok := collectionsByName[schema.Name]
+		if !ok {
+			return NewErrAddCollectionWithPatch(schema.Name)
+		}
+
+		col, err := db.newCollection(collectionDescription, schema)
 		if err != nil {
 			return err
 		}
 
 		newCollections = append(newCollections, col)
-		newSchemaByName[col.schema.Name] = col.schema
 	}
 
 	for i, col := range newCollections {
 		col, err := db.updateCollection(
```

Comment:
todo: Discussed via a video call but noting here. We should rename this function to `updateSchema`.

Reply:
resolving as it's been noted on the function itself.